src/core/library.c

```diff
@@ -663,6 +663,49 @@ QuicLibraryGetGlobalParam(
         Status = QUIC_STATUS_SUCCESS;
         break;
 
+    case QUIC_PARAM_GLOBAL_PERF_COUNTERS: {
+
+        if (*BufferLength < sizeof(uint64_t)) {
+            *BufferLength = sizeof(uint64_t) * QUIC_PERF_COUNTER_MAX;
+            Status = QUIC_STATUS_BUFFER_TOO_SMALL;
+            break;
+        }
+
+        if (Buffer == NULL) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+
+        if (*BufferLength < QUIC_PERF_COUNTER_MAX * sizeof(uint64_t)) {
+            //
+            // Copy as many counters will fit completely in the buffer.
+            //
+            *BufferLength = (*BufferLength / sizeof(uint64_t)) * sizeof(uint64_t);
+        } else {
+            *BufferLength = QUIC_PERF_COUNTER_MAX * sizeof(uint64_t);
+        }
+
+        uint64_t* Counters = (uint64_t*)Buffer;
+        memcpy(Buffer, MsQuicLib.PerProc[0].PerfCounters, *BufferLength);
+
+        for (int ProcIndex = 1; ProcIndex < MsQuicLib.ProcessorCount; ++ProcIndex) {
+            for (int CounterIndex = 0;
+                CounterIndex < (*BufferLength / sizeof(uint64_t));
+                ++CounterIndex) {
+                Counters[CounterIndex] =
+                    ((int64_t)Counters[CounterIndex]) +
+                    (int64_t)MsQuicLib.PerProc[ProcIndex].PerfCounters[CounterIndex];
+            }
+        }
+
+        if((int64_t)Counters[QUIC_PERF_COUNTER_CONN_ACTIVE] < 0) {
+            Counters[QUIC_PERF_COUNTER_CONN_ACTIVE] = 0;
+        }
+
+        Status = QUIC_STATUS_SUCCESS;
+        break;
+        }
```

Comment:
```suggestion
    }
```

Reply:
Doesn't this confuse with the closing brace for the enclosing `switch` statement?